src/pymatgen/core/structure.py

```diff
@@ -3338,6 +3339,29 @@ def to_conventional(self, **kwargs) -> Structure:
         """
         return self.to_cell("conventional", **kwargs)
 
+    def get_moyo_dataset(self, **kwargs) -> moyopy.MoyoDataset:
```

Comment:
From the method name, it is unclear it is a symmetry related function. I would name it get_moyo_symmetry_dataset. Alternatively, why not just put the same functionality in get_symmetry_dataset and have a new kwarg called `use_moyo=False`.

Reply:
`get_moyo_symmetry_dataset` sounds good.

> Alternatively, why not just put the same functionality in get_symmetry_dataset and have a new kwarg called use_moyo=False.

i'm reluctant to do that as you could no longer have a single return type and so you would give up on auto-complete suggestions in your IDE:

![Screenshot 2025-01-27 at 12 13 12](https://github.com/user-attachments/assets/1ee64d19-d669-4007-a17c-c0207b07aa96)

Reply:
besides the better auto-complete UX, directly returning the `Moyo/SpglibDataset` also has the benefit of reducing maintenance burden. when new keys and symmetry results are added to `moyopy` in the future, they don't have to be translated from `MoyoDataset` to `dict` format

Reply:
If we have one method name for both backends, we would need to make sure the return value has some kind of useful type (eg some minimal dataclass) and not just a bare dict, otherwise I agree the maintenance burden would be too high when the return values inevitably become inconsistent between the different backends.

Reply:
This could include converting the spglib dict into the `MoyoDataset` format if it seems appropriate, but we really should just have one type returned.

Reply:
Ok, we take on the maintenance burden. We should still have a good idea of exactly what is being returned here _especially_ if there are multiple backends.

As a proposal, how about we have `get_symmetry_dataset -> moyopy.MoyoDataset | spglib.SpglibDataset` as suggested, but we  use [`typing.overload`](https://docs.python.org/3/library/typing.html#typing.overload) to ensure everything is typed nicely?

Then we get the benefit of the general method name, the multiple backends, and the typing being correct.